### PR TITLE
Add `StrictFactory`

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

### CODE REVIEW

**Comment by samdark** [comment]
It's good implementation-wise. What's the use case?

**Comment by vjik** [comment]
> It's good implementation-wise. What's the use case?

I want to try to use it in Yii Queue refactoring: https://github.com/yiisoft/queue/pull/222

File: README.md [hunk 1]
```diff
@@ -73,6 +73,29 @@ turn it off by passing `false` as a third constructor argument:
 $factory = new Factory($container, $factoryConfig, false);
 ```
 
+### Strict factory
+
+`StrictFactory` is similar to base factory, but creating objects for specified definitions only:
```
Comment (roxblnfk) [inline README.md:78]: `StrictFactory` differs in that it processes only configured definitions.

Comment (roxblnfk) [inline README.md:78]:
```suggestion
`StrictFactory` differs in that it processes only configured definitions.
When attempting to request an existing class that is not defined in the factory config,
a `NotFoundException` will be thrown.
```

**Review by samdark — APPROVED** [review]

**Pull request merged**